GenumKit/Stencil/CallNode.swift

```diff
@@ -0,0 +1,92 @@
+//
+// GenumKit
+// Copyright (c) 2016 Olivier Halligon
+// MIT Licence
+//
+
+import Stencil
+
+struct CallableBlock {
+  let parameters: [String]
+  let nodes: [NodeType]
+
+  func context(_ context: Context, arguments: [Variable]) throws -> [String: Any] {
+    var result = [String: Any]()
+
+    for (parameter, argument) in zip(parameters, arguments) {
+      result[parameter] = try argument.resolve(context)
+    }
+
+    return result
+  }
+}
+
+open class DefineNode: NodeType {
+  open let variableName: String
+  open let parameters: [String]
+  open let nodes: [NodeType]
+
+  open class func parse(_ parser: TokenParser, token: Token) throws -> NodeType {
+    let components = token.components()
+    guard components.count >= 2 else {
+      throw TemplateSyntaxError("'define' tag takes at least one argument, the name of the variable to set")
+    }
+    let variable = components[1]
+    let parameters = Array(components.dropFirst(2))
+
+    let setNodes = try parser.parse(until(["enddefine"]))
+    guard parser.nextToken() != nil else {
+      throw TemplateSyntaxError("`enddefine` was not found.")
+    }
+
+    return DefineNode(variableName: variable, parameters: parameters, nodes: setNodes)
+  }
+
+  public init(variableName: String, parameters: [String], nodes: [NodeType]) {
+    self.variableName = variableName
+    self.parameters = parameters
+    self.nodes = nodes
+  }
+
+  open func render(_ context: Context) throws -> String {
+    let result = CallableBlock(parameters: parameters, nodes: nodes)
+    context[variableName] = result
```

Comment:
Do you think we should keep it that way or would it be better to fail for `enumBlock` and render the `enumBlock2`?